src/web/index.php

```diff
@@ -1,28 +1,174 @@
 <?php
+
+const PAGINATING_INDEX = 5;
+
 require_once './functions.php';
 
 $airports = require './airports.php';
 
+$filterLetter = $_GET['filter_by_first_letter'] ?? null;
+$filterSort = $_GET['sort'] ?? null;
+$filterPage = $_GET['page'] ?? 1;
+$filterState = $_GET['filter_by_state'] ?? null;
+$filtersArr = [$filterLetter, $filterState, $filterSort];
+
 // Filtering
 /**
  * Here you need to check $_GET request if it has any filtering
  * and apply filtering by First Airport Name Letter and/or Airport State
  * (see Filtering tasks 1 and 2 below)
+
+ * @param string $filterLetter
+ * @param array $airports
+ * @return array
+ */
+function firstLetterFilter(string $filterLetter, array $airports): array
+{
+
+    $filteredArr = [];
+
+    foreach ($airports as $airport) {
+        if (mb_substr($airport['name'], 0, 1) === $filterLetter) {
+            $filteredArr[] = $airport;
+        }
+    }
+
+    return $filteredArr;
+
+}
+
+/**
+ * @param string $filterState
+ * @param array $airports
+ * @return array
  */
+function stateFilter(string $filterState, array $airports): array
+{
+
+    $filteredArr = [];
+
+    foreach ($airports as $airport) {
+        if ($airport['state'] === $filterState) {
+            $filteredArr[] = $airport;
+        }
+    }
+
+    return $filteredArr;
+
+}
+
+if ($filterLetter) {
+    $airports = firstLetterFilter($filterLetter, $airports);
+}
+
+if ($filterState) {
+    $airports = stateFilter($filterState, $airports);
+}
 
 // Sorting
 /**
  * Here you need to check $_GET request if it has sorting key
  * and apply sorting
  * (see Sorting task below)
+
+ * @param string $sortKey
+ * @param array $airports
+ * @return array
  */
+function codeSorting(string $sortKey, array $airports): array
+{
+
+    $unFilteredArr = $airports;
+    $sortKeyArr = array_column($airports, $sortKey);
+    $ind = array_multisort($sortKeyArr, SORT_ASC, $airports);
```

Comment:
Maybe you can find a more readable name for the variable?

Reply:
Changed "unFilteredArr" to "inputArray" and "sortKeyArr" to "keysArray" in last coomit